tensorflow/python/keras/utils/vis_utils_test.py

```diff
@@ -103,6 +104,129 @@ def test_plot_model_with_add_loss(self):
     except ImportError:
       pass
 
+  def test_dot_layer_range(self):
+    model = efficientnet.EfficientNetB0()
+
+    # Case 1: when layer names in normal (increasing) order
+    layer_range = ['block1a_project_conv', 'block1a_activation']
+    layer_ids_from_model = get_layer_ids_from_model(
+      model, layer_range
+    )
+    try:
+      dot = vis_utils.model_to_dot(model, layer_range=layer_range)
+      dot_edges = dot.get_edges()
+      layer_ids_from_dot = get_layer_ids_from_dot(
+        dot_edges
+      )
+      self.assertAllEqual(sorted(layer_ids_from_model),
+        sorted(layer_ids_from_dot))
+    except ImportError:
+      pass
+
+    # Case 2: when layer names in abnormal (decreasing) order
+    layer_range = ['block1a_activation', 'block1a_project_conv']
+    layer_ids_from_model = get_layer_ids_from_model(
+      model, layer_range
+    )
+    try:
+      dot = vis_utils.model_to_dot(model, layer_range=layer_range)
+      dot_edges = dot.get_edges()
+      layer_ids_from_dot = get_layer_ids_from_dot(
+        dot_edges
+      )
+      self.assertAllEqual(sorted(layer_ids_from_model),
+        sorted(layer_ids_from_dot))
+    except ImportError:
+      pass
+
+  def test_plot_layer_range(self):
+    model = efficientnet.EfficientNetB0()
+
+    # Case 1: when layer names in normal (increasing) order
+    layer_range = ['block1a_project_conv', 'block1a_activation']
+    effnet_subplot = 'model_effnet_1.png'
+    try:
+      vis_utils.plot_model(model,
+        to_file=effnet_subplot,
+        layer_range=layer_range)
+      self.assertTrue(file_io.file_exists_v2(effnet_subplot))
+      file_io.delete_file_v2(effnet_subplot)
+    except ImportError:
+      pass
+
+    # Case 2: when layer names in abnormal (decreasing) order
+    layer_range = ['block1a_activation', 'block1a_project_conv']
+    effnet_subplot = 'model_effnet_2.png'
+    try:
+      vis_utils.plot_model(model,
+        to_file=effnet_subplot,
+        layer_range=layer_range)
+      self.assertTrue(file_io.file_exists_v2(effnet_subplot))
+      file_io.delete_file_v2(effnet_subplot)
+    except ImportError:
+      pass
+
+  def test_layer_range_fails(self):
+    model = efficientnet.EfficientNetB0()
+
+    try:
+
+      # Case 1: When layer_range don't have complete subgraph.
+      with self.assertRaises(AssertionError):
```

Comment:
Since all the test logic is very same, please convert this to a parameterized test to avoid all the duplications.

Reply:
done

Reply:
cool. This is much cleaner.